drivers/usb/device/usb_dc_rpi_pico.c

```diff
@@ -121,6 +121,14 @@ static int udc_rpi_start_xfer(uint8_t ep, const void *data, size_t len)
 
 	ep_state->next_pid ^= 1u;
 	*ep_state->buf_ctl = val;
+	/*
+	 * By default, clk_sys runs at 125MHz, wait 3 nop instructions before
+	 * setting the AVAILABLE bit. See 4.1.2.5.1. Concurrent access.
+	 */
+	arch_nop();
+	arch_nop();
+	arch_nop();
+	*ep_state->buf_ctl |= USB_BUF_CTRL_AVAIL;
```

Comment:
```suggestion
*ep_state->but_ctl = val | USB_BUF_CTRL_AVAIL;
```

There is no point in re-reading the value (`buf_ctl` is volatile).